### ISSUE TITLE
How can I animate the route to show a pulsating trail?

### ISSUE THREAD

**Issue by fzkhan19** [visgl/react-google-maps#387]
### Description

I want to animate the route displayed using direction renderer, I think deck.gl might help in this but im not sure how. i want a functionality same as show in this video: https://www.youtube.com/watch?v=H_BWRTf0d8g

### Steps to Reproduce

none

### Environment

- Library version:
- Google maps version: weekly
- Browser and Version:
- OS:

**Comment by usefulthink** [comment]
yep, I don't think something like that can be done with the vanilla DirectionsRenderer that comes with the Maps JavaScript API. I think the best choice for this would be the deck.gl [TripsLayer](https://deck.gl/docs/api-reference/geo-layers/trips-layer) (see also the demo here: https://deck.gl/examples/trips-layer). 

For that you need to specify the points along the path together with timestamps and then animate the 'currentTime' value of the layer. If you don't have a 'true' timing reference you can use e.g. the relative distance along the path as timestamp. 

That TripsLayer can then be added to the map with an interleaved overlay as shown here: https://visgl.github.io/react-google-maps/docs/guides/deckgl-integration#using-googlemapsoverlay

**Comment by fzkhan19** [comment]
I  tried to write the code, now i'm struggling to get it to animate
```js
```

**Comment by fzkhan19** [comment]
I  tried to write the code, now i'm struggling to get it to animate
```js

```

**Comment by fzkhan19** [comment]
its working now there was some problem with loop length,
now I'm having issues with default center and default zoom, they don't seem to be working

**Comment by usefulthink** [comment]
I believe the default behavior of the DirectionsRenderer is to fit the map to the bounds of the route. 
Also, what do you need the DirectionsRenderer for if you're rendering the Route using the TripsLayer?
